config-xml/addon/src/main/java/org/jboss/windup/config/parser/XMLLabelProviderLoader.java

```diff
@@ -0,0 +1,191 @@
+package org.jboss.windup.config.parser;
+
+import org.jboss.forge.furnace.Furnace;
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.windup.config.LabelProvider;
+import org.jboss.windup.config.loader.LabelProviderLoader;
+import org.jboss.windup.config.loader.RuleLoaderContext;
+import org.jboss.windup.config.metadata.LabelMetadataBuilder;
+import org.jboss.windup.config.metadata.LabelProviderMetadata;
+import org.jboss.windup.util.Logging;
+import org.jboss.windup.util.exception.WindupException;
+import org.jboss.windup.util.furnace.FileExtensionFilter;
+import org.jboss.windup.util.furnace.FurnaceClasspathScanner;
+import org.w3c.dom.Document;
+
+import javax.inject.Inject;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.*;
+import java.util.logging.Logger;
+
+/**
```

Comment:
Please add comments to describe the class and the methods as well.

Reply:
done